File: 08.BrainTwist脑筋急转弯/leetcode_657_judgeCircle.py
/bin/bash: line 1: :w: command not found

class Solution:
    def judgeCircle(self, moves: str) -> bool:
        cnt = Counter(moves)
        return cnt['R']==cnt['L'] and cnt['U']==cnt['D']

if __name__ == "__main__":
    s = Solution()
    moves = "UD"
    print(s.judgeCircle(moves))

File: 08.BrainTwist脑筋急转弯/leetcode_657_judgeCircle.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool judgeCircle(string moves) {
        unordered_map<char, int> cnt;
        for (auto& c : moves){
            cnt[c]++;
        }
        return cnt['R']==cnt['L'] && cnt['U']==cnt['D'];
    }
};

int main(){
    Solution s;
    string moves = "UD";
    cout << s.judgeCircle(moves) << endl;
}
